internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -77,7 +84,7 @@ var clusterSchema = map[string]*schema.Schema{
 var clusterSpecSchema = &schema.Schema{
 	Type:        schema.TypeList,
 	Description: "Spec for the cluster",
-	Required:    true,
+	Optional:    true,
```

Comment:
How come the spec of cluster is optional? Can you create the cluster without any spec fields?

Reply:
As we are using the same Schema for data source and resource, I had to make the spec optional.

This is the same pattern as is followed in other resources. I can create a new schema if this is not appropriate.